### DISCUSSION TITLE
Скопировать определенные строки

### DISCUSSION THREAD

**Discussion by Murbulat1** [totumonline/totum-ru-issues-and-discussions#367] [Как это сделать?] [answered]
Доброго времени суток. Начинаю изучать ТОТУМ, это что то с чем то(в хорошем смысле ^^)

Есть таблица "А", в нем достаточно много значении, самый основной это **статус**.
Так же есть таблица "Б", где хотелось бы реализовать вывод строк с таблицы "А" но только те строки, где в статусе указан определеное значение. 

`Т.е. если в таблице А есть две строки, и у одного из них в статусе указан (процессе [3]) а у второго (готовый[1]), та строка где в процессе должен будет отобразится на таблице Б`

Чего я хочу добится, если значение статуса в таблице "А" поменять на другую, то с таблицы "Б" это запись автоматически должна будет исчезнуть.

На фото, пример того чтобы строки со статусом "Выполнен", дублировались в "Счет" (это как пример)
![3](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/92213625/df223890-92e4-43dd-b830-1c515e0d5ec3)

**Comment by Murbulat1** [comment]
Проштудировал документацию от TOTUM , но видимо я не достаточно опытен чтобы сообразить что и как....

**Comment by erpic-ru** [comment, marked as answer]
Поизучайте временные таблицы и все, что с ними связано. Например, [это видео](https://www.youtube.com/watch?v=fHeg1hlXIqQ). Возможно, и не нужно ничего копировать

**Reply by ttm-alexeypolunin** [reply]
Временные таблицы и загрузка туда необходимых строк один из вариантов.

Если вам необходимо сделать Фильтр, те что бы строки были именно эти (из таблицы А), то можно сделать фильтр в дереве:

https://ru.docs.totum.online/system-tables#stables-tree

<img width="1677" alt="Снимок экрана 2023-05-12 в 15 04 58" src="https://github.com/totumonline/totum-ru-issues-and-discussions/assets/55755565/23acce04-35a0-445c-b3b4-c6bc67799071">

**Reply by ttm-alexeypolunin** [reply]
Если необходимо, что бы у вас было реально две параллельные таблицы и из второй необходимо строку добавлять/изменять/удалять (такое бывает иногда нужно), то это делается кодами-действий: https://ru.docs.totum.online/action-codes-base-level

**Reply by Murbulat1** [reply]
Здравствуйте, прочитал про дерево, попытался сделать так как описано... Возможно я что то делаю не так? 
Добавил в дерево новую ветку, с типом ярлык, настроил фильтр.
![2](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/92213625/515caf1a-91c2-441c-abee-e224a69219cf)

Но когда открывается все так же все записи со всеми статусами...
![4](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/92213625/40c4120a-ac51-4969-8717-685a82235782)

Я же правильно понимаю, перейти по созданному ярлыку из дерева, то должно быть по умолчанию от фильтрованный список по полю?

Так же ознакомился с временными таблицами еще раз, думал я что то пропустил но нет.. @erpic-ru спасибо за то что сказали куда копать ^^ Временная таблица в моем случае не подходит, мне необходимо из основной таблицы А1 вывести только те записи в таблицу Б1 где статус будет = 1 (как пример).


@ttm-alexeypolunin , спасибо за ответ

**Reply by ttm-alexeypolunin** [reply]
Ярлыки работают вместе с полями Префильтров. Те в основной таблице нужно создать поле Префильтра по статусу и ярлыком передавать в него значение. Поле должно быть доступно для редактирования пользователю, но может быть скрыто по видимости.

Доки по префильтрам: https://ru.docs.totum.online/prefilters-basic-level
Префильтры по Селектам: https://ru.docs.totum.online/how-to-filter-by-select

**Reply by erpic-ru** [reply]
Так при автозаполнении временной таблицы можно выбрать ограничение выборки в коде через where "status" = "нужное значение" и собрать только нужные записи
